javascript/ql/src/Security/CWE-094/MethodNameInjection.qhelp

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>
+Invoking a user-controlled method on certain objects can lead to invocation of unsafe functions,
+such as <code>eval</code> or the <code>Function</code> constructor. In particular, the global object
+contains the <code>eval</code> function, and any function object contains the <code>Function</code> constructor
+in its <code>constructor</code> property.
+</p>
+</overview>
+
+<recommendation>
+<p>
+Avoid invoking user-controlled methods on the global object or on any function object.
+Whitelist the permitted method names or change the type of object the methods are stored on.
+</p>
+</recommendation>
+
+<example>
+<p>
+In the following example, a message from the document's parent frame can invoke the <code>play</code>
+or <code>pause</code> method. However, it can also invoke <code>eval</code>.
+A malicious website could embed the page in an iframe and execute arbitrary code by sending a message
+with the name <code>eval</code>.
+</p>
+
+<sample src="examples/MethodNameInjection.js" />
+
+<p>
+Instead of storing the API methods in the global scope, put them in an API object. It is also good
```

Comment:
We could also suggest using a `Map` instead of a plain object.

Reply:
Added `Map` to the qhelp file, although I didn't use it in the example as I found the difference in style to be distracting from what the example is showing.

For example, if we replace `function play()` with:
```javascript
api.put("play", (data) => {
  // ...
});
```
any manual calls to the `play` function would have to be changed to `api.get("play")(...)` which isn't really an attractive option.